include/pybind11/detail/common.h

```diff
@@ -9,6 +9,11 @@
 
 #pragma once
 
+#define PYBIND11_VERSION_MAJOR 2
+#define PYBIND11_VERSION_MINOR 6
+#define PYBIND11_VERSION_PATCH 0
+#define PYBIND11_VERSION_TYPE "dev"
```

Comment:
I guess it makes sense that we're now working on 2.6.0, but shouldn't we leave this to @wjakob, to do all the release and version number acrobatics?

Reply:
I had to replace `PYBIND11_VERSION_PATCH dev1` with something. Until it's released, this shouldn't matter too much. I could have done 2.5.1, or 2.5.0 dev, it didn't matter, I just needed something with an integer in PATCH.

Reply:
Fine with me. But just to clarify: we can no longer put `dev` into the patch definition here? That will maybe need some clarifications for `docs/release.rst`.

Reply:
One potential issue as well: many of the places that reference `PYBIND11_VERSION_MAJOR` etc. to generate a combined version string now do not concatenate that with `PYBIND11_VERSION_TYPE`.

Reply:
Or `dev<number>`, sorry. I'm used to [PEP 440](https://www.python.org/dev/peps/pep-0440/) numbering, but this seems to already be just a "dev" attached to a normal PATCH number, which is easy to parse in CMake.

Reply:
It would actually be nice to notice in a CMake build when you're not building with an officially released version, though? Or not?

Reply:
I've thought about that. But you don't get the benefit of an automatic "dev" tag if you are not using CMake, then.

Reply:
I've restored the old C++ (except for moving the lines up a bit), CMake now parses patch into path and type. We could change this later (as currently, if someone were to do a comparison on the patch version in a macro, it won't work for the dev version - not that that would be common), but that's can be addressed later if desired.